students_documentation/MLJ_model_report.md

```diff
@@ -0,0 +1,125 @@
+- Though `MLJBase.jl` offers a `predict()` method that works for any model from any MLJ-supported library and the `probs()` method required by the `AbstractFittedModel` type can be implemented using this `predict()` method, the package doesn't offer a similar method that could be used to implement the `logits()` method, which is also required by the `AbstractFittedModel` type. Thus, methods from the specific library providing an interface to a particular MLJ model have to be called in order to retrieve the logits for that model. This is, again, unreasonable to do inside a one-size-fits-all interface.
+- Even if it was possible to effectively differentiate between regression, classification and unsupervised models, and also possible to implement a general `logits()` method that works for all MLJ models, it still seems necessary to differentiate between different MLJ models within our package, since different models have different compatibility with the counterfactual generators. For example, there are some models that are only compatible with gradient-based generators, some that are only compatible with the Feature Tweak generator, and some that are compatible with both gradient-based generators and Feature Tweak. It seems most reasonable to enforce this separation between the models by implementing support for them through separate interfaces.
+
+
+### BetaML.jl
+
+| **Model** | **Analysis** |
+| -------- | ------- |
+| DecisionTreeClassifier, RandomForestClassifier | Currently incompatible with the gradient-based generators implemented in our package. However, counterfactual explanations can be generated for these models using the Feature Tweak generator we implemented. The models could also be made compatible with gradient-based generators in the future, possibly through the use of [probability calibration](https://scikit-learn.org/stable/modules/calibration.html). |
+| NeuralNetworkClassifier, LinearPerceptron | Currently incompatible with all generators. However, it should be possible to make these models compatible with all gradient-based generators once the best way to access their gradients has been figured out: all gradient-based generators need gradient-access and currently leverage `Zygote.jl` for that, but `Zygote.jl` does not work for MLJ models, so an alternative solution has to be found. |
+| KernelPerceptron | Currently incompatible with all generators, since the model is not differentiable. The possibility of using probability calibration should be explored. |
```

Comment:
Can't we use FeatureTweak for this one?

Reply:
Assuming you mean KernelPerceptron - I think not, Feature Tweak only works on trees as far as I know.

Reply:
Yeah, that's what I meant. Okay, thanks for the clarification!